signer/xpi/x509.go

```diff
@@ -108,8 +108,8 @@ func (s *XPISigner) makeTemplate(cn string) *x509.Certificate {
 			Province:           []string{"CA"},
 			Locality:           []string{"Mountain View"},
 		},
-		NotBefore:          EENotBefore,
-		NotAfter:           EENotBefore.Add(87600 * time.Hour), // ten year
+		NotBefore:          time.Now().UTC(),
```

Comment:
Is there any chance for a race condition here? Like should it be `now() - 5 seconds` or similar?

Reply:
Good question! Which events are you worried about racing?

I believe the expected order of events is:

1. addon signed / NotBefore generated
1. (optional) addon installed (possibly multiple installs)
1. blocklist updated. This assumes it knows about all addons signed at the time (ideally I think we'd use the PK7 SigningTime attr but that's a client side change)

I don't think we'd run into problems backdating by five seconds, but there's a chance that a concurrent backdated addon signing and a blocklist update could generate a blocklist that doesn't include the backdated NotBefore time.

Generally, backdating certs is frowned upon (AFAIK it's a violation of the web CA baseline requirements), but we do backdate content signatures to account for client clock skew so if there's an edge case we're missing it'd be good to consider that.

Reply:
Ok totally! With that order, it makes sense that it is extremely unlikely for there to be a race condition. As well, this was more of a general curiosity rather than me being specifically concerned.

Reply:
Got it! I appreciate having to think it through.

As we chatted about we'll have to be careful with unit tests that sign and verify quickly since those could race depending on the resolution the verifier uses.